### PR TITLE
Implement jitResolveFlattenableField

### PR DESCRIPTION
**Pull request by gacholio**
Also some trivial cleanup.

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

File: runtime/codert_vm/cnathelp.cpp [hunk 1]
```diff
@@ -2409,10 +2409,36 @@ void* J9FASTCALL
...
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
 	DECLARE_JIT_PARM(J9Method*, method, 1);
 	DECLARE_JIT_INT_PARM(cpIndex, 2);
 	DECLARE_JIT_INT_PARM(resolveType, 3);
-	void *addr = NULL;
+	J9ConstantPool * const ramConstantPool = J9_CP_FROM_METHOD(method);
+	J9RAMFieldRef * const ramFieldRef = ((J9RAMFieldRef *)ramConstantPool) + cpIndex;
+	UDATA const flags = ramFieldRef->flags;
+	UDATA const valueOffset = ramFieldRef->valueOffset;
+	if (!VM_VMHelpers::instanceFieldRefIsResolved(flags, valueOffset)) {
```
Comment (tajila) [inline runtime/codert_vm/cnathelp.cpp:2422]: do we need an additional check to ensure its resolved for final fields if `J9TR_FLAT_RESOLVE_PUTFIELD == resolveFlags` ?

Comment (gacholio) [reply runtime/codert_vm/cnathelp.cpp:2422]: I copied the checks and resolve flags from the bytecodes - would anything more than that be required?

Comment (tajila) [reply runtime/codert_vm/cnathelp.cpp:2422]: I was wondering about this

```
			/* Ref is resolved, see if it's fully resolved for put (put resolved and not a final field) */
			UDATA const resolvedBit = J9FieldFlagPutResolved;
			UDATA const finalBit = J9AccFinal;
			UDATA const testBits = resolvedBit | finalBit;
			UDATA const bits = flags & testBits;
			/* Put resolved for a non-final field means fully resolved */
			if (J9_UNEXPECTED(resolvedBit != bits)) {
				/* If not put resolved for a final field, resolve is necessary */
				if (J9_UNEXPECTED(testBits != bits)) {
					goto resolve;
				}
				/* Final field - ensure the running method is allowed to store */
				if (J9_UNEXPECTED(!J9ROMMETHOD_ALLOW_FINAL_FIELD_WRITES(J9_ROM_METHOD_FROM_RAM_METHOD(_literals), 0))) {
					if (J9_UNEXPECTED(VM_VMHelpers::ramClassChecksFinalStores(ramConstantPool->ramClass))) {
						/* Store not allowed - run the resolve code to throw the exception */
						goto resolve;
					}
				}
			}
```

If the fieldRef was resolved for a getfield, do we need another pass?

Comment (gacholio) [reply runtime/codert_vm/cnathelp.cpp:2422]: Yep missed that, I'll work on including this.

Comment (gacholio) [reply runtime/codert_vm/cnathelp.cpp]: Fixed.

**Review by tajila — APPROVED** [review]

**Comment by tajila** [comment]
Jenkins test sanity,extended xlinuxval jdknext

**Comment by tajila** [comment]
Jenkins test sanity win jdk8,jdk11

**Comment by tajila** [comment]
FYI @hzongaro

**Pull request merged**